### PR TITLE
Inline the value of `%r` in `T_FMT`

### PR DESCRIPTION
**Pull request by pitdicker**
Similar to `D_T_FMT`, `T_FMT` can refer to `T_FMT_AMPM` with `%r`.
In plain English: a locale can say the default way to format a time is a 12-hour clock with AM/PM, using whatever is defined for that format in that locale.

I should have noticed this when working on https://github.com/chronotope/pure-rust-locales/pull/5, sorry :disappointed:.

Fixes https://github.com/chronotope/chrono/issues/1349 when we update chrono to the new version.

Can this become a new release? I picked 0.8.0 as this changes the data.

### CODE REVIEW

**Comment by pitdicker** [comment]
Sorry, found one more weirdness. Converting to draft for now.

**Comment by pitdicker** [comment]
Actually found two more cases of tricky locale data :disappointed:.

* In some locales `T_FMT_AMPM` exists but is empty, while there is a value for `AM_PM`. In that case we should treat it the same as when `T_FMT_AMPM` does not exists and set the value to the POSIX default of `%l:%M:%S %p`.
* In two locales `T_FMT` refers to `T_FMT_AMPM`, but `T_FMT_AMPM` and `AM_PM` are empty. Here I picked `%T` as fallback, the default value for a 24-hour clock.

I did an extra sanity check of the data and didn't find other issues (for now).

**Review by cecton — APPROVED** [review]
oops! :sweat_smile: I didn't notice!

Feel free to rewrite everything or part of it if you think you can improve readability/maintainability. I'm not sole owner anymore. (In fact I'm not sure if I'm owner at all, I'm just here to help : )

**Comment by pitdicker** [comment]
Thank you for sharing the crate and the review!

@djc Do you want to have a look? And what do you think about a new release?

File: generate-api/src/main.rs [hunk 1]
```diff
@@ -97,11 +97,22 @@ fn validate_and_fix_t_fmt_ampm(objects: &mut Vec<Object>) {
...
         if !found_t_fmt_ampm {
+            // Insert `T_FMT_AMPM` if it is missing
             let value = match am_pm_empty {
                 true => vec![Value::String(String::new())],
                 false => vec![Value::String("%l:%M:%S %p".to_string())],
             };
             object.values.push(("t_fmt_ampm".to_string(), value));
+        } else if !am_pm_empty {
+            // Update `T_FMT_AMPM` if it is empty and we do have a value for `AM_PM`.
+            for (key, ref mut value) in object.values.iter_mut() {
```
[the remark this answers is not in the document]
Comment (pitdicker) [reply generate-api/src/main.rs]: No, good catch.

We should set up a Clippy job on the CI and fix all the suggestions. I'd like to leave that for a later PR.

File: generate-api/src/main.rs [hunk 2]
```diff
@@ -141,7 +142,7 @@ fn validate_and_fix_d_t_fmt(objects: &mut Vec<Object>) {
         }
         for (key, ref mut value) in object.values.iter_mut() {
             match (key.as_str(), value) {
-                ("d_t_fmt", vec) => {
+                ("d_t_fmt", vec) | ("t_fmt", vec) => {
```
[the remark this answers is not in the document]
Comment (pitdicker) [reply generate-api/src/main.rs:145]: No, every commit does. But because of https://github.com/chronotope/pure-rust-locales/issues/13 they pack a 2,5mb file with every commit.

**Comment by pitdicker** [comment]
> For me each of these commits is quite opaque. It would be good if the commit had some example change in terms of the expected end-user effect.

Yes, it would be easier to see if every commit contained a diff with the changes to the generated data.

> Update `T_FMT_AMPM` if it is empty and we do have a value for `AM_PM`.

The first example is in `af_ZA`. A format string with `%r` currently will come up with no time in the output string because `T_FMT_AMPM` exists but has no value. This inserts the default POSIX value like when `T_FMT_AMPM` is completely absent.

An alternative is to detect this case in chrono at runtime. Because we are fixing up the data anyway this seemed the cleaner solution.

> Fall back to `%T` if `%r` points to an empty `T_FMT_AMPM`

This effects two locales, `hy_AM` and `sm_WS`. Again the result in chrono is no time in the output string if `T_FMT` or `D_T_FMT` (`%X` or `%c`) specify a 12-hour clock with `%r` while apparently not having a 12-hour clock format.

> Inline the value of `%r` in `T_FMT`

Going from one level of locale string in `T_FMT` down another level in `T_FMT_AMPM` is tricky. We inlined the value in `D_T_FMT` to enable parsing a format string without allocating, but I missed `T_FMT`.

**Review by djc — APPROVED** [review]
Nice!

**Comment by pitdicker** [comment]
@cecton Can I help with making the release?

**Comment by pitdicker** [comment]
> @cecton Can I help with making the release?

@cecton Friendly ping.

**Comment by cecton** [comment]
Oh I'm sorry!! I misunderstood you.

I see you don't have ownership on the crate. I'm adding you and @djc

**Comment by cecton** [comment]
I just did cargo-release in the past but feel free to upgrade the process

**Comment by pitdicker** [comment]
Thank you very much! Published 0.8.0.
